### ISSUE TITLE
subtypes(Function) hangs

### ISSUE THREAD

**Issue by gaoxinge** [JuliaLang/julia#29947]
## platform

 - win10
-  julia 1.0.1

## problem

If run
```
julia> subtypes(Function)
```
in repl, the command will consume a long time.

- Does this conform to expectation?

**Comment by StefanKarpinski** [comment]
Yes, better to ask these kinds of questions on discourse.

**Comment by JeffBezanson** [comment]
Fixed by #30705; should be in 1.1.

**Comment by simonbyrne** [comment]
I'm still seeing it on 1.1

**Comment by JeffBezanson** [comment]
Ah, maybe that PR missed the branch?

**Comment by diegozea** [comment]
It is fixed in 1.3.0-DEV.115 but not in 1.1:

Julia 1.3.0-DEV.115
```
julia> @elapsed subtypes(Function)
0.832130617
```
Julia 1.1
```
julia> @elapsed subtypes(Function)
369.488036327
```

**Comment by KristofferC** [comment]
Yes, this was said a few comments up.
